src/main/daml/Daml/Finance/Holding/Fungible.daml

```diff
@@ -39,11 +40,11 @@ template Fungible
     signatory account.custodian, account.owner, getLockers this
     observer Disclosure.flattenObservers observers
 
-    ensure amount > 0.0
+    ensure amount > 0.0 && not (isSome lock && S.null (getLockers this))
```

Comment:
Curious as to why the check for empty lockers isn't directly on the interface `Acquire` choice of the `Holding.Base`?

Reply:
Interfaces currently don't support `ensure`, and all choice bodies of interfaces are provided by methods (effectively deferring all checks to the implementation). 

An alternative would be to let the `newLockers : Parties` of the `Acquire` choice be a non-empty set.

Reply:
> Interfaces currently don't support `ensure`, and all choice bodies of interfaces are provided by methods (effectively deferring all checks to the implementation).

Concretely, what I mean is this :

```haskell
choice Acquire : ContractId Base
    -- ^ Lock a contract.
    with
      newLockers : Parties
        -- ^ Parties which restrain the contract's ability to perform specified actions.
      context : Text
        -- ^ Reason for acquiring a lock.
      lockType : LockType
        -- ^ Type of lock to acquire
    controller (view this).account.owner, newLockers
    do
      assert (not $ S.null newLockers) "Lockers set must contain at least one party"
      acquire this arg
```

Reply:
I know, but we decided to keep all our choice bodies "naked" (meaning they consist of an interface method - here `acquire` - effectively deferring all such checks to the implementation).

Reply:
> I know, but we decided to keep all our choice bodies "naked" (meaning they consist of an interface method - here `acquire` - effectively deferring all such checks to the implementation).

I see - it does mean that the error message that a failing `ensure` clause throws isn't going to provide any context into why it's failing other than that it has failed. Ie., no error message no reason as to why it has failed (could be due to the `amount` being 0.0 or less; or due to the empty lockers).

🤷

Reply:
If you look at the `acquireImpl`, that check is now in there. The additional `ensure` check has the advantage that a `Holding` can't be created with a `lock` with empty `newLockers`.

Reply:
That makes sense. 

Another approach would be that if the contract is locked with the empty set of `lockers`, then allow the account owner to `Release` the lock?

```haskell
  choice Release : ContractId Base
    -- ^ Unlock a locked contract.
    with
      context : Text
    controller getLockers this
    do
      release this arg

-- | Get the lockers of a holding.
getLockers : (HasToInterface t Base) => t -> Parties
getLockers holding = optional mempty (.lockers) (view $ toInterface @Base holding).lock
``` 

^^ if `.lockers` is empty, return the `account.owner` as a set of parties.

Reply:
Yes, that would work. However, the `owner` could lock the holding with itself as `lockers` and achieve the same behaviour. I would argue that is more explicit. 

I think it is easier to understand if `lockers` are non-empty sets.

Reply:
It's more of a 'get out of jail' if the `owner` makes a mistake creates a contract with `lockers` set as null to avoid the `ensure` clause which every implementation now has to added. 

I certainly agree with you that being explicit is better than implicit...